### PR TITLE
select, insert, update, delete methods for category table (need to refactor)

### PR DESCRIPTION
**Pull request by Resistendi**

### CODE REVIEW

File: database/helpers.py [hunk 1]
```diff
@@ -88,3 +95,63 @@ def _convert_resultproxy_to_dictionary(result_proxy):
...
+async def add_limit(title: str, limit: float, start_date: datetime, period: int, is_repeated: bool) -> None:
+    args = {
+        'title': title,
+        'limit': limit,
+        'start_date': start_date,
+        'period': period,
+        'is_repeated': is_repeated
+    }
+    category = Category.__table__
+    ins = category.insert().values(args)
```
Comment (IrinaLokhvytska) [inline database/helpers.py]: Use from sqlalchemy import update, insert to category with title that already exists cause an error. Or add check `check_category_exists`: if it exists - update, if not - insert. Or you can use upsert - http://docs.sqlalchemy.org/en/latest/dialects/postgresql.html#insert-on-conflict-upsert

File: database/helpers.py [hunk 2]
```diff
@@ -88,3 +95,63 @@ def _convert_resultproxy_to_dictionary(result_proxy):
...
+    category = Category.__table__
+    upd = category.update().where(Category.title == category_name).values(kwargs)
+    engine = cren(dsn)
+    conn = engine.connect()
+    conn.execute(upd)
+
+
+async def delete_limit(category_name: str):
+    category = Category.__table__
+    delete = category.delete().where(Category.title == category_name)
```
Comment (IrinaLokhvytska) [inline database/helpers.py]: Don't delete category, just set limit to zero or Null.

File: database/helpers.py [hunk 3]
```diff
@@ -1,11 +1,18 @@
 import psycopg2
+from datetime import datetime
 from aiopg.sa import create_engine
 from config import db
 from sqlalchemy.schema import CreateTable, DropTable
+from sqlalchemy import select
 from database.models import Transaction
+from database.models import Category
+from sqlalchemy import create_engine as cren
```
Comment (sapphireua) [inline database/helpers.py]: Please sort imports due to https://www.python.org/dev/peps/pep-0328/ . If you use `Pycharm`, you can execute `Optimize imports` command (https://www.jetbrains.com/help/pycharm/creating-and-optimizing-imports.html)

File: database/helpers.py [hunk 4]
```diff
@@ -39,7 +46,7 @@
 
 async def _prepare_tables():
     """Asynchronous function for creating tables in database."""
-    tables = [Transaction.__table__]
+    tables = [Category.__table__, Transaction.__table__]
```
Comment (sapphireua) [inline database/helpers.py]: Here a suggest to use some kind of dynamic tables population, something like this https://github.com/evgeniyDorogoy/moneyfier/blob/master/database/helper.py#L28

File: database/helpers.py [hunk 5]
```diff
@@ -88,3 +95,63 @@ def _convert_resultproxy_to_dictionary(result_proxy):
     for row in result_proxy:
         dict_result.append(dict(row))
     return dict_result
+
+
+async def get_limit(category_name=None) -> list:
```
Comment (sapphireua) [inline database/helpers.py]: If you function return list of dictionaries, typing shoul lokks like `-> list[dict]` https://docs.python.org/3/library/typing.html

Comment (Resistendi) [reply database/helpers.py]: when i'm using -> list[dict] its throwing TypeError: 'type' object is not subscriptable

Comment (sapphireua) [reply database/helpers.py]: Here is a decision:

```
from typing import Dict, Tuple, List

async def get_limit(category_name=None) -> List[Dict]:
```

**Pull request closed without merging**